### PR TITLE
Added conditional support to RenderShroud/JammerCircle

### LINKED ISSUE #17333: RenderCircle type traits arent conditional
RenderShroudCircle:
RenderJammerCircle:
are not conditional, 
this makes it not possible to: 
 -display Mobile Gap Generator gap generation radius when its inside an naval transport.
 -display Mobile Radar Jammer jamming radius when its inside an naval transport.
They should able to accept an RequiresCondition: so that i can pass an [actor id]: [condition] given inside an PassengerConditions: of an Cargo: to them for rendering the appropriate circle type of given actor inside.

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/OpenRA/issues/17333

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Render/RenderShroudCircle.cs [hunk 1]
```diff
@@ -58,17 +61,18 @@ public IEnumerable<IRenderable> RenderAnnotations(WorldRenderer wr, World w, Act
...
 		WDist range;
 
 		public RenderShroudCircle(RenderShroudCircleInfo info)
+			: base(info)
 		{
 			this.info = info;
 		}
 
-		void INotifyCreated.Created(Actor self)
+		protected override void Created(Actor self)
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Traits/Render/RenderShroudCircle.cs:75]: Don't you also need to call `base.Created(self);`

**Review by PunkPun — DISMISSED** [review]
LGTM

**Review by pchote — APPROVED** [review]
LGTM, removed the testcase commit so we can merge.

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e488f0edbb1d3cac5ad7bea97b2c905a74c14d14

**Pull request merged**